src/components/app/LoadStoredProfileDataManager.js

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+// This component is responsible for caching the stored profile data in the
+// redux state.  This will control whether we can delete this profile.
+
+// Implementation note:
+// This is done as a separate component than where the deletion button is
+// rendered, so that the button doesn't flick, because the access to the stored
+// profile data is asynchronous.
+// Also this isn't in a thunk action to make it independent from any other
+// change. It only depends on the current hash being changed.
+
+import { PureComponent } from 'react';
+
+import { getHash } from 'firefox-profiler/selectors/url-state';
+import { cacheStoredProfileData } from 'firefox-profiler/actions/app';
+
+import { retrieveProfileData } from 'firefox-profiler/app-logic/published-profiles-store';
+import explicitConnect from 'firefox-profiler/utils/connect';
+
+import type { ConnectedProps } from 'firefox-profiler/utils/connect';
+
+type StateProps = {|
+  +hash: string,
+|};
+
+type DispatchProps = {|
+  +cacheStoredProfileData: typeof cacheStoredProfileData,
+|};
+
+type Props = ConnectedProps<{||}, StateProps, DispatchProps>;
+
+class LoadStoredProfileDataManagerImpl extends PureComponent<Props> {
```

Comment:
This feels like a lot of effort with an entire component here, but I don't really have context on the problem.

Reply:
This is an entire component, but otherwise a simple one.

I think this is pretty standard React programming to trigger an action following a state change in an untied way.